packages/electron-updater/src/AppUpdater.ts

```diff
@@ -514,15 +522,20 @@ export abstract class AppUpdater extends (EventEmitter as new () => TypedEmitter
     }
 
     try {
-      return this.doDownloadUpdate({
+      this.downloadPromise = this.doDownloadUpdate({
         updateInfoAndProvider,
         requestHeaders: this.computeRequestHeaders(updateInfoAndProvider.provider),
         cancellationToken,
         disableWebInstaller: this.disableWebInstaller,
         disableDifferentialDownload: this.disableDifferentialDownload,
-      }).catch((e: any) => {
-        throw errorHandler(e)
       })
+        .catch((e: any) => {
+          throw errorHandler(e)
+        })
+        .finally(() => {
+          this.downloadPromise = null
+        })
+      return this.downloadPromise
     } catch (e: any) {
       return Promise.reject(errorHandler(e))
```

Comment:
I think we need to add `this.downloadPromise = null` before this `return` as well as a failsafe.

Reply:
It's already been added in the finally block, so it doesn't need to be added here.

Reply:
Oh interesting. The `finally` block still executes even when we throw inside of the `catch`?

Reply:
Yes. Though I just found out that the error we throw on catch is not caught afterwards. The thrown error in a promise can only be caught in the promise chain i.e. the try-catch block won't catch it.

```js
try {
    Promise.reject("err")
        .catch((e) => {
            console.log("caught in promise");
            throw e;
        })
        .finally(() => console.log("finally promise"))
} catch (error) {
    console.log("caught in try-catch")
}
```

Regardless the try-catch block is not needed.